Source/WebKit/gtk/migrating-to-webkitgtk-6.0.md

```diff
@@ -88,12 +91,30 @@ APIs have been moved from [type@WebKit.WebContext] and [type@WebKit.WebsiteDataM
 [type@WebKit.NetworkSession]. There's a default global persistent session that you can get with
 [func@WebKit.NetworkSession.get_default]. You can also create new sessions with
 [ctor@WebKit.NetworkSession.new] for persistent sessions and [ctor@WebKit.NetworkSession.new_ephemeral]
-for ephemeral sessions. It's no longer possible to create a [type@WebKit.WebsiteDataManager], it's now
+for ephemeral sessions. It's no longer possible to create a [type@WebKit.WebsiteDataManager]; it's now
 created by the [type@WebKit.NetworkSession] automatically at construction time. The [type@WebKit.NetworkSession]
 to be used must be passed to the [type@WebKit.WebView] as a construct parameter. You can pass the
 same [type@WebKit.NetworkSession] object to several web views to use the same session. The only exception
 is automation mode, which uses its own ephemeral session that is configured by the automation
-session capabilities.
+session capabilities. If you notice that your application uses [type@WebKit.WebContext] or
+[type@WebKit.WebsiteDataManager] APIs that no longer exist, look for replacement APIs
+in [type@WebKit.NetworkSession].
+
+`webkit_web_context_clear_cache()` does not have a direct replacement, but you
+can use [method@WebKit.WebsiteDataManager.clear] to achieve the same effect.
+
+The `WebKitWebContext::download-started` signal has been removed. Use
+[signal@WebKit.NetworkSession::download-started] instead.
+
+## Favicon Database
+
+[type@WebKit.FaviconDatabase] is now owned by [type@WebKit.WebsiteDataManager]
+instead of [type@WebKit.WebContext], and favicons are now enabled by default.
+Use [method@WebKit.WebsiteDataManager.set_favicons_enabled] if you do not need
+favicons.
```

Comment:
Hm, maybe we should not enable favicon database by default, because favicons are not used for any purpose unless the application does something manually with WebKitFaviconDatabase....

Reply:
There's `WebKitWebView:favicon` property, you don't need to use `WebKitFaviconDatabase` at all, just monitor favicon property and use `webkit_web_view_get_favicon` to get the favicon. I agree with not enabling them by default, though.

Reply:
https://bugs.webkit.org/show_bug.cgi?id=251588